src/ntttcp.h

```diff
@@ -34,6 +34,8 @@ struct ntttcp_test
 	uint	server_ports; /*    How many ports opening in receiver side, from -m flag, or -P flag */
 	int	cpu_affinity; /*    CPU affinity, from -m flag */
 	char	*bind_address; /*    Socket binding address */
+	bool	source_addr_bind; /* '-S' option for tx side */
+	char	*source_address;  /* NIC Interface address */
```

Comment:
You could use source_address == NULL to denote absence of the option, and remove the bool flag.